services/web-server/src/utils/unpromisify.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Wrap an async `fn` in such a way that it can act as a function taking a callback
+ * as its last argument.  That callback either gets cb(err) for errors, or cb(null, res)
+ * or (if returnsArray is true) cb(null, ...res)
+ */
+module.exports = (fn, {returnsArray} = {}) => {
+  // eslint-disable-next-line no-unused-vars
+  const call = (self, args) => {
+    const done = args.pop();
+    Promise.resolve(fn.apply(self, args)).then(
+      returnsArray ? res => done.call(null, null, ...res) : res => done.call(null, null, res),
+      err => done.call(null, err));
+  };
+
+  // preserve arity by generating code..
+  const args = [...new Array(fn.length + 1)].map((_, i) => `arg${i}`);
+  // eslint-disable-next-line no-eval
+  return eval(`const x = function (${args}) { return call(this, [...arguments]); }; x`);
```

Comment:
It looks like the only reason why we are using `eval` is so that we can spread the list of arguments. I think we should be able to avoid this by doing something like:

```js
return (args) => {
  return call(this, [...args]);
}(args);
```

Thoughts?

Reply:
No, see the comment above about arity.  The code you've written has arity 1.

Reply:
Sorry, I meant to write:

```js
return (...args) => {
  return call(this, args);
}(args);
```

Reply:
Sorry, I meant to write:

```js
return (...args) => {
  return call(this, [...args]);
}(args);
```

Reply:
That has arity 0 :)

Reply:
You're right. My mistake.

Reply:
I wrestled with this stuff for a while writing this patch!